Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -41,9 +43,41 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         desc.add<std::vector<std::string> >("filelist", std::vector<std::string>({}))
             ->setComment("list of files with the readout cells of each module");
         desc.add<edm::ESInputTag>("cellindexer", edm::ESInputTag(""))->setComment("Dense cell index tool");
+        desc.add<std::string>("offsetfile", std::string({}))
+            ->setComment("file containing the offsets between calibration and surrounding cells");
         descriptions.addWithDefaultLabel(desc);
       }
 
+      std::map<std::tuple<std::string, int, int>, int> makeOffsetMap(std::string input_offsetfile) {
```

Comment:
This `std::map<std::tuple<std::string, int, int>, int>` looks like a rather inefficient approach.

If the map is going to be accessed event by event, can you check what the actual cost of using it ?

To be fair, the whole `produce()` method looks pretty bad from the performance point of view. Anything that uses string matching should simply be avoided.

Reply:
> If the map is going to be accessed event by event, can you check what the actual cost of using it ?

This is an ESProducer, so the map would be created and accessed once per IOV, so it isn't _that_ bad, but possibly some optimization could be worth it anyway.

Reply:
Ah, OK, the map itself is not part of the payload.